Framework/PythonInterface/plugins/algorithms/GenerateLogbook.py

```diff
@@ -0,0 +1,356 @@
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid import config
+from mantid.api import AlgorithmFactory, FileAction, FileProperty, \
+    ITableWorkspaceProperty, Progress, PythonAlgorithm
+from mantid.kernel import Direction, IntArrayOrderedPairsValidator, \
+    StringListValidator
+from mantid.simpleapi import *
+
+import fnmatch
+import h5py
+import numpy
+import os
+import re
+
+
+class GenerateLogbook(PythonAlgorithm):
+
+    _data_directory = None
+    _facility = None
+    _instrument = None
+    _numor_range = None
+    _metadata_headers = None
+    _metadata_entries = None
+
+    def category(self):
+        return 'Utility'
+
+    def summary(self):
+        return 'Generates logbook containing meta-data specific to the instrument and technique used to obtain the raw data'
+
+    def name(self):
+        return 'GenerateLogbook'
+
+    def validateInputs(self):
+        issues = dict()
+
+        if self.getProperty('Instrument').isDefault:
+            issues['Instrument'] = 'The instrument has to be defined.'
+        else:
+            instrument = self.getPropertyValue('Instrument')
+            ws_tmp = CreateSingleValuedWorkspace()
+            try:
+                LoadParameterFile(Workspace=ws_tmp, Filename=instrument + '_Parameters.xml')
+            except Exception as e:
+                self.log().error(str(e))
+                issues['Instrument'] = 'There is no parameter file for {} instrument.'.format(instrument)
+            DeleteWorkspace(Workspace=ws_tmp)
+
+        if not self.getProperty('CustomEntries').isDefault:
+            custom_entries = self.getPropertyValue('CustomEntries')
+            custom_entries = custom_entries.split(',')
+            if not self.getProperty('CustomHeaders').isDefault:
+                custom_headers = self.getPropertyValue('CustomHeaders')
+                custom_headers = custom_headers.split(',')
+                if len(custom_entries) != len(custom_headers):
+                    issues['CustomHeaders'] = 'Provide none or as many headers as custom entries.'
+
+        return issues
+
+    def PyInit(self):
+
+        self.declareProperty(FileProperty('Directory', '',
+                                          action=FileAction.Directory),
+                             doc='Path to directory containing data files for logging.')
+
+        self.declareProperty(ITableWorkspaceProperty('OutputWorkspace', '',
+                                                     direction=Direction.Output),
+                             doc='The output table workspace.')
+
+        self.declareProperty("NumorRange", [0, 0],
+                             direction=Direction.Input,
+                             validator=IntArrayOrderedPairsValidator(),
```

Comment:
I indeed have come across MFP multiple times, and I am not convinced it would be the best choice here. First, it would start searching for the provided numors in the user search directories, not only in the desired directory which can cause confusion unless it is specified as an alternative workflow for this algorithm. Second, the validation of large number of files may be slow, as suggested by issue #29881. Also, the very convenient, in other circumstances, workspace operations possible with MFP would need to be disallowed. 

I am going to check how fast it can deal with a couple of thousand of files on a remote drive, and if it slows down the execution I would rather not use it.

Reply:
If you've noted that the MFP type exists and have decide not to use it that's fine - just wanted to double check. If you decide not to use it then I still think it's worth using : for the range so people used to specifying ranges in the MFP can use the same syntax here for a range

Reply:
I replaced `IntArrayOrderedPairsValidator` with `IntArrayBoundedValidator`, and with a couple of lines of custom validation it provides the desired functionality, where the user can specify a numor range using a colon or a dash, along with giving two boundary values.